Sources/Routing/Router.swift

```diff
@@ -18,22 +22,46 @@ public class Router {
     /// - parameter output: the associated output of this path, usually a responder, or `nil`
     public func register(host: String?, method: HTTP.Method, path: [String], responder: Responder) {
         let host = host ?? "*"
-        let path = [host, method.description] + path.filter { !$0.isEmpty }
-        base.extend(path, output: responder)
+        let path = path.filter{ !$0.isEmpty }
+        
+        // Registers `route` metadata
+        let route = Route(host: host, components: path, method: method)
+        routeMetadata.append(route)
+        
+        let fullPath = [host, method.description] + path
+        base.extend(fullPath, output: responder)
+    }
+    
+    public func register(
+        host: String?,
+        method: HTTP.Method,
+        path: [String],
+        metadata: [String: Any],
+        responder: Responder
+        ) {
+        let host = host ?? "*"
+        let path = path.filter{ !$0.isEmpty }
+        
+        // Registers `route` metadata
+        let route = Route(host: host, components: path, method: method, metadata: metadata)
+        routeMetadata.append(route)
+        
+        let fullPath = [host, method.description] + path
+        base.extend(fullPath, output: responder)
     }
     
     // caches static route resolutions
     private var _cache: [String: Responder?] = [:]
     private var _cacheLock = NSLock()
-
+    
```

Comment:
If you head into your git client you can unstage these specific lines you've changed, which is a bit more complex in a CLI. A commit containing only a whitespace change isn't really "clean". That's for next time 👍

Reply:
Oops!